Lesson16/Lesson16-Busra/busra.js

```diff
@@ -0,0 +1,442 @@
+    - Define a function `findMin(numbers)` that loops through the array
+      to find and return the smallest number.
+    - Log the smallest number.
+*/
+const MyNumbersEx17 = [4,9,2,1,7,15,3,11];
+
+
+function findMin(numbers){
+  let smallestNumber = numbers[0];
+  for(let i=0; i < numbers.length; i++){
+    if(numbers[i] < smallestNumber){
+      smallestNumber= numbers[i];
+    }
+  }
+  return smallestNumber;
+}
+findMin(MyNumbersEx17);
+
+
+
+/*
+18. Count Occurrences of a Word in an Array
+    - Define a function `countOccurrences(arr, word)` that loops through `arr`
+      to count how many times `word` appears.
+    - Log the count.
+
+*/
+
+const foods=["apple","egg","banana","apple","bread","apple"];
+
+
+function countOccurrences(arr, word){
+  let counterEx18 = 0;
+    for(let i =0; i < arr.length; i++){
+        if(arr[i].toLowerCase() === word.toLowerCase()){
+          counterEx18 +=1;
+        }
+    }
+    return counterEx18;
+}
+
+
+countOccurrences(foods,"apple");
+
+
+/*
+19. Remove Falsy Values
+    - Define a function `removeFalsyValues(arr)` that loops through an array
+      and returns a new array without falsy values (false, 0, "", null, undefined, NaN).
+    - Log the new array.
+*/
+const myArrayEx19=["string", 3 , "number", 8, false, 0 ,"", null, undefined, NaN];
+
+function removeFalsyValues(arr){
+   let myNewArrayEx19=[];
+   for(let i=0; i < arr.length; i++){
+    if(arr[i]){
+      myNewArrayEx19.push(arr[i]);
+    }
+    return myNewArrayEx19;
+   }
+}
+removeFalsyValues(myArrayEx19);
+
+
+
+/*
+20. Sum of All Digits in a String
+    - Define a function `sumDigits(str)` that loops through each character of `str`,
+      checks if it's a digit, and if so, adds it to a total sum.
+    - Log the final sum.
+    - Example: "abc123" -> 6
+
+*/
+const strEx20 = "def432ne379";
+function sumDigits(str){
+  let countEx20 = 0;
+  for(let i=0; i<str.length; i++){
+    if(!isNaN(str[i]) && str[i] !== ' '){
+      counter += Number(str[i]);
+    }
+  }
+  return counter;
+}
+
+sumDigits(strEx20);
+
+
+/*
+21. Average of Array Elements
+    - Define a function `averageArray(numbers)` that uses a loop
+      to calculate the average (sum / length).
+    - Log the average.
+*/
+
+const myArrayEx21 = [5,7,10,520,789,320,2,84];
+function averageArray(numbers){
+  let sumEx21 = 0;
+  for(let i=0; i<numbers.length; i++){
+    sumEx21 = sumEx21 + Number(numbers[i]);
+  }
+  let AvEx21 = sumEx21 / numbers.length;
+  return AvEx21;
+}
+
+
+averageArray(myArrayEx21);
+
+
+/*
+22. Flatten a 2D Array (Nested Loops)
+    - Define a function `flattenArray(twoDArray)` that takes an array of arrays
+      (e.g., [[1,2],[3,4]]) and uses nested loops to create a new one-dimensional array.
+    - Log the flattened array.
+*/
+
+const My2DArray = [[1,4,7],[2,5,8]];
+
+
+function flattenArray(twoDArray){
+  let my1DArray =[];
+  for(let i=0; i<twoDArray.length; i++){
+    for(let j=0; i< twoDArray[i].length; j++){
```

Comment:
This works if the array only contain arrays as its elements. Can you try to solve this for these example as well:`[[1,2,3], 2, true, [1, 2, 3]];`?